src/main/cpp/src/decimal_utils.cu

```diff
@@ -696,13 +711,15 @@ private:
 };
 
 // Functor to divide two DECIMAL128 columns with rounding and overflow detection.
+template <typename T>
 struct dec128_divider : public thrust::unary_function<cudf::size_type, __int128_t> {
```

Comment:
I just realized that I made a big mistake doing the processing this way and I need to fix this. We should not be doing a thrust::transform.  I am going to put up a PR and see if I can fix this.

Reply:
Okay I put up a PR for this fix

https://github.com/NVIDIA/spark-rapids-jni/pull/820

please take a look and upmerge as needed.